Lib/fontTools/ttLib/tables/otBase.py

```diff
@@ -654,9 +654,15 @@ def decompile(self, reader, font):
 
 	def compile(self, writer, font):
 		self.ensureDecompiled()
+		# TODO Following hack to be removed by rewriting how FormatSwitching tables
+		# are handled.
+		# https://github.com/fonttools/fonttools/pull/2238#issuecomment-805192631
 		if hasattr(self, 'preWrite'):
+			deleteFormat = not hasattr(self, 'Format')
 			table = self.preWrite(font)
+			deleteFormat = deleteFormat and hasattr(self, 'Format')
```

Comment:
What does this double check do?

Reply:
it means "delete Format attribute if it wasn't there before calling preWrite and it's been added by the preWrite"

Reply:
So why delete it at the end of the function?

Reply:
it's used when serializing the tables, we only want to delete after compile is completed

Reply:
It's just a hack. Ideally compilation shouldn't modify the objects. The `preWrite` methods are modifying by adding .Format. Eventually we will remove that, by rewriting how format-switching table machinery works. But for now, to make it work, I let `preWrite` add .Format, but will delete it at the end of compile when they are not needed anymore.

Another place we modify objects during compilation is when we split large lookups. But that's for another day...